### DISCUSSION TITLE
Suggestion: Add timeout to bot actions

### DISCUSSION THREAD

**Discussion by AnalogKnight** [FakeFishGames/Barotrauma#10822] [Ideas and suggestions]
We all know that bots sometimes get stuck on some actions. Such as repairing the top docking hatch, reloading the drone, etc. This can often cause serious problems in single player games. For example, the sole engineer on the submarine got stuck trying to fix the electrical equipment on the drone due to waypoint issues. Players may not notice this immediately until the sub loses power, by which time it may be too late.
Solving the waypoint problem might not be an easy job. But adding a "timeout" state timer to the bot might easily solve this problem. For example, when the bot tries to repair the top docking hatch and gets stuck on the ladder due to a waypoint problem, if after more than 30 seconds the bot realizes that the action is still not complete, it will give up and move on to the next possible task, Such as repairing other damaged equipment. It would be even better if the game could alert the player by marking devices that cannot be repaired on the screen.

**Comment by Kyr4l** [comment]
this is a good idea
that would work around many of the issues that bots have, even if it's not a permanent fix to some sisues, it's better than nothing
